### PR TITLE
Should never synchronize on a Boolean

### PR DESCRIPTION
**Pull request by adamretter**
A Boolean only has two static instances True or False, and they are different instances depending on the value, and so synchronising on a Boolean actually leads to synchronising on one or two global instances which is not what was intended.

### CODE REVIEW

File: exist-core/src/main/java/org/exist/storage/BrokerPool.java [hunk 1]
```diff
@@ -975,29 +974,25 @@ public NotificationService getNotificationService() {
...
     public boolean isReadOnly() {
-        synchronized(readOnly) {
-            if(!readOnly) {
-                final long freeSpace = FileUtils.measureFileStore(dataLock.getFile(), FileStore::getUsableSpace);
-                if (freeSpace != -1 && freeSpace < diskSpaceMin) {
-                    LOG.fatal("Partition containing DATA_DIR: {} is running out of disk space [minimum: {} free: {}]. Switching eXist-db into read-only mode to prevent data loss!", dataLock.getFile().toAbsolutePath().toString(), diskSpaceMin, freeSpace);
-                    setReadOnly();
-                }
+        final boolean isReadOnly = readOnly.get();
+        if (!isReadOnly) {
```
Comment (reinhapa) [inline exist-core/src/main/java/org/exist/storage/BrokerPool.java:983]: I think this local variable is only used here and can be inlined

**Review by reinhapa — APPROVED** [review]

File: exist-core/src/main/java/org/exist/storage/BrokerPool.java (hunk 1 above)
Comment (adamretter) [reply exist-core/src/main/java/org/exist/storage/BrokerPool.java:983]: I felt it was clearer to read in the way I formulated it. How strong is your objection?

Comment (dizzzz) [reply exist-core/src/main/java/org/exist/storage/BrokerPool.java:983]: for debugging is is more clear; I'd not care

**Review by dizzzz — APPROVED** [review]

File: exist-core/src/main/java/org/exist/storage/BrokerPool.java (hunk 1 above)
Comment (reinhapa) [reply exist-core/src/main/java/org/exist/storage/BrokerPool.java:983]: @adamretter just a hint, not a requirement..

**Pull request merged**